<html>
<head>
<title>
Tao Jiang
</title>
</head>

<body>

<table border=0 cellpadding=15>
<tr align=top>
<td align=top>
<img src="./TaoGetty13a.jpg"
ismap width=180 height=200>
</td>
<td align=top> <br>
<b> Tao Jiang </b> <br> 
Distinguished Professor <br>
<a href="http://www.cs.ucr.edu/">
Department of Computer Science and Engineering <br>
</a>
<a href="http://www.ucr.edu/">
University of California - Riverside <br>
</a>
Riverside, CA 92521 <br>
lastname AT cs DOT ucr DOT edu <br>
Phone: (951) 827-2991 <br>
Fax: (951) 827-4643 <br>
Office:  Winston Chung Hall (WCH), Room 336 <br>
<br>
</td>
</tr>
</table>
<p>
<!-- ********************************************************************** -->




<hr>
I received B.S. in Computer Science and Technology from the 
<a href=http://www.ustc.edu.cn/> 
University of Science and Technology of China (USTC) </a>,
Hefei, P.R.China in July 1984 and Ph.D. in Computer Science
from the <a href=http://www1.umn.edu/twincities/> University of Minnesota </a> in Nov. 1988.
During Jan. 1989 - June 2001, I was a faculty member at
<a href=http://www.cas.mcmaster.ca> Department of Computing and Software</a>,
McMaster University, Hamilton, Ontario, Canada.
I joined <a href=http://www.ucr.edu> University of
California - Riverside </a> as Professor of Computer Science in 
Sept. 1999, and am a member of the
<a href=http://algorithms.cs.ucr.edu/>Algorithms and Computational Biology Lab</a>, 
<a href=http://www.genetics.ucr.edu>Genetics Graduate Program</a>,
<a href=http://www.genomics.ucr.edu/>Institute for Integrative Genome Biology</a>,
and
<a href=http://www.cepceb.ucr.edu/>Center for Plant Cell Biology</a>.
I also held visiting positions at
<a href=http://bioinfo.au.tsinghua.edu.cn/en>Tsinghua University</a>
and 
<a href=http://www.scbit.org/scbit_en/pages/index.do>Shanghai Center for Bioinformation
Technology (SCBIT)</a>. 

<p>
For a complete list of my publications and other detailed info, please see my 
<a href=https://scholar.google.com/citations?user=XUhsCZwAAAAJ>Google Scholar page</a>
or <a href="resume.pdf">CV</a> or <a href="biography-short.txt">short biography</a>.
I have recently served on the editorial boards of 
<a href=http://www.informatik.uni-trier.de/~ley/db/journals/jco/> 
Journal of Combinatorial Optimization (JCO) </a>, 
<a href=http://www.biomedcentral.com/bmcbioinformatics/> BMC Bioinformatics</a>
and
<a href=http://www.springerlink.com/content/100117/> Algorithmica</a>,
and program committees of 
<a href=http://recomb2023.bilkent.edu.tr/>RECOMB'2023</a> and
<a href=https://www.iscb.org/ismbeccb2023>ISMB/ECCB'2023</a>.


<h2> 
Research
</h2>

<ul>
<li> Design and analysis of algorithms.

<p>We study discrete objects such as strings, trees, graphs, etc.,
and have a special interest in the design of efficient approximation
algorithms with good performance bounds.
Our past work includes approximation algorithms for 
shortest common superstrings and directed Steiner trees.
We have also worked on an average-case analysis technique based on the
incompressibility method, which involves the theory of Kolmogorov complexity.
Our result on the average size of 
<a href="http://mathworld.wolfram.com/HeilbronnTriangleProblem.html">
Heilbronn's triangles</a>
was featured/reported in popular science magazines and newspapers:

<UL>
<LI> <a href="http://www.newscientist.com/">New Scientist</a>, Nov. 6, 1999, pp. 44-47, "On a Roll" by Dana Mackenzie.

<LI> <a href="http://www.ams.org/mathmedia/mathdigest/">Math Digest</a>, Dec. 1999.

<LI> <a href="http://www.courrierinternational.com/">Courrier International</a>,
Dec. 23, 1999 -- Jan. 5, 2000 (#41), pp. 477-478:
"Le hasard ne joue pas aux de's", par Dana Mackenzie (translation of the NS
article). 

<LI> <a href="http://www.zeit.de/">Die Zeit</a>, 
April 13, 2000 (#16), p. 40: "Geometrisch Eingekreist" by Wolfgang Blum.

</UL> 
<p> 

Our lower bound on the average-case complexity of the well-known Shellsort algorithm
is mentioned on this <a href=https://en.wikipedia.org/wiki/Shellsort>wiki page</a>, 
and still remains as the best!

<li> Computational molecular biology.

<p> We are interested in developing efficient algorithms and software
for computational problems in molecular biology and genomics.
Our past work includes efficient algorithms for
haplotype inference on pedigrees, genome-scale ortholog assignment, 
isoform inference, metagenomics, and isoform-based differential analysis.
<a href=http://engr.case.edu/li_jing/haplotyping.html>PedPhase</a> is
a suite of programs for inferring haplotypes from genotypes on
a pedigree.
<a href=http://tishunter.ucr.edu/>TISHunter</a> is
a web server for predicting translation initiation sites in eukaryotic mRNAs
using a support vector machine.  
<a href=http://msoar.cs.ucr.edu/>MSOAR</a> is a method for predicting
orthologous genes between closely related genomes that takes gene order into
consideration (see the <a href=https://github.com/fishinucr/msoar2.0>Github page</a>
for source code).
<a href=http://www.cs.ucr.edu/~liw/isolasso.html>IsoLasso</a> is a combinatorial
approach for inferring mRNA isoforms and their expression levels from RNA-Seq data.
See the 
<a href=http://www.genengnews.com/gen-articles/gene-expression-enters-transformative-times/5066/>
feature article </a> in Genetic Engineering and Biotechnology News 
mentioning our work on isoform inference and differential expression analysis.

<p> I have extensively collaborated with 
<a href=http://www.sladeklab.ucr.edu/aboutpi.html>Frances Sladek</a>,
<a href=http://faculty.ucr.edu/~tgirke/girke.html>Thomas Girke</a>,
and 
<a href=http://plantpath.ucr.edu/people/faculty/borneman.html>James Borneman</a>.
Our research was recently funded by 
<a href=http://www.nsf.gov/awardsearch/showAward?AWD_ID=1646333>NSF EAGER</a>,
<a href=http://www.nsf.gov/awardsearch/showAward?AWD_ID=1262107>NSF ABI</a>
(joint with Grace Xiao at UCLA; see the
<a href=http://www.nsf.gov/awardsearch/showAward?AWD_ID=1262134> collaborative
project</a>)
and
<a href=http://projectreporter.nih.gov/project_info_description.cfm?aid=8258935&icde=12339560&ddparam=&ddvalue=&ddsub=&cr=4&csb=default&cs=ASC>NIH NIDDK</a> (joint with Frances Sladek),
Here is a (quite outdated) <A HREF="jiang-lab-overview.ppt">overview</A> of 
some of the projects in my lab. 


<p> If you are interested in learning the abc of computational biology
and bioinformatics, you may find some useful
<a href=http://www.colorbasepair.com/bioinformatics_courses_tutorials.html>
educational material</a> on the subject (in particular,
<a href=http://www.animalgenome.org/edu/doe/index.html> A Primer on Molecular Genetics</a>). 
Our book <a href=https://mitpress.mit.edu/books/current-topics-computational-molecular-biology>
Current Topics in Computational Molecular Biology</a> was
    published by the MIT Press as a part of its Computational Molecular
    Biology Series (and co-published by Tsinghua Univ.
    Press in China) although the topics covered in the book are probably 
    no longer current in the field :-)

<!--<p>-->
<!--We are currently looking for an outstanding postdoc to work on our-->
<!--computational biology projects. The announcement of the position-->
<!--can be found <A HREF="bioinfo-pos.txt">here</A>.-->

<p>
<li> Complexity of finite automata based models.
<p>
The following are two examples of long-standing open problems that we are
interested in: (i) succinctness of two-way (nondeterministic and deterministic) 
finite automata and (ii) detection of connectivity in digital images by 
2-dimensional finite automata. Kolmogorov complexity has been the main tool 
in our past work on similar problems.

</ul>
<h2> 
Teaching
</h2>

<ul>
I am teaching one undergraduate and two graduate courses in the
winter and spring quarters this year.

<p>
<li> <b> CS 141 </b>: Intermediate Data Structrues and Algorithms,
<a href=http://www.cs.ucr.edu/~jiang/141-homepage.html>winter quarter</a>,
2009.

<p>
<li> <b> CS 238 </b>: Algorithmic Techniques in Computational Biology, 
<a href=http://www.cs.ucr.edu/~jiang/238-homepage.html>spring quarter</a>,
2025.
We examine some fundamental algorithmic topics in computational
molecular biology through instructor lecturing, student presentation,
and assigned reading and discussion.

<p>
<li> <b> CS 150 </b>: The Theory of Automata and Formal Languages,
<a href=http://www.cs.ucr.edu/~jiang/150-homepage.html>spring quarter</a>, 2025.

<p>
<li> <b> CS 260 </b>: Seminar: Selected Topics in Bioinformatics, 
<a href=http://www.cs.ucr.edu/~jiang/260-homepage.html>winter quarter</a>,
2019.
This class mainly consists of student presentations of their own
research and interesting work from the current bioinformatics and genomics
literature. 

<!--<p>-->
<!-- <li> <b> Ph.D Depth (Qualifier) Exam </b>: Theory of Computation and Algorithms.-->
<!-- You may find a copy of the -->
<!-- <a href="csexams/Qual-announce07.pdf"> syllabus of the exam </a>  -->
<!-- and <a href="csexams/Qualifier06.pdf"> -->
<!-- the 2006 depth exam </a> in PDF format. You can also find many previous comprehensives -->
<!-- in Prof. Chrobak's public directory ~marek/PUBLIC. -->
<!-- Students who are interested in taking the exam should send Prof. Lonardi -->
<!-- or Prof. Chrobak or Prof. Young or me a message as early as possible. -->

<p>
<li> <b> Seminar in Algorithms and Bioinformatics (SAB)</b>: a weekly
seminar series where students present their research work and interesting
papers from recent literature. Its schedule can be found at
<a href="sab-seminar.txt">this page</a>. 

<!--<p>-->
<!-- Here is one of the top 5 -->
<!-- <a href="cs2md3/Othello.html"> Othello programs </a> -->
<!-- done by students in my data structures and algorithms class as their -->
<!-- programming project -->
<!-- in 1997. The program is written in Java, by M. Plug, D. Willits, and B. Vujic. -->
<!-- See if you can beat it :-) -->

</ul>

<p>

<h2> 
Office Hours
</h2>
MW 10-11am in W20, but please always feel free to drop in.
<!-- MW 2-3pm in W16, but please always feel free to drop in. -->

<h2> 
Former Postdocs and Ph.D Students
</h2>
<li> <a href=http://www.cs.cityu.edu.hk/~lwang/>Lusheng Wang</a>
<li> <a href=http://www.cs.uic.edu/~dasgupta/>Bhaskar DasGupta</a>
<li> <a href=http://www.cs.mun.ca/~harold/>Todd Wareham</a>
<li> <a href=https://gianluca.dellavedova.org/>Gianluca Della Vedova</a>
<li> <a href=http://www.cs.ualberta.ca/~ghlin/>Guohui Lin</a>
<li> <a href=https://scholar.google.at/citations?user=HI4v4ZIAAAAJ&hl=en>Claus Vogl</a>
<li> <a href=http://webs.wichita.edu/?u=bioscience&p=/people/faculty/lJia/index>Li Jia</a>
<li> <a href=https://sites.google.com/a/case.edu/jingli/about/lab-director>Jing Li</a>
<li> <a href=http://www.utrgv.edu/csci/faculty/andres-figueroa/index.htm>Andres Figueroa</a>
<li> <a href=http://kam.mff.cuni.cz/~kolman/>Petr Kolman</a> 
<li> <a href=https://www.researchgate.net/profile/Xin_Chen78>Xin Chen</a>
<li> <a href=http://www.cs.ucr.edu/~hli/>Haifeng Li</a> (now at Motorola)
<li> <a href=https://shanghaitechzhengjielab.github.io/Zhenglab.github.io/index.htm>Jie Zheng</a> (joint with Stefano Lonardi)
<li> <a href=https://www.linkedin.com/in/chuhu-yang-77807515>Chuhu Yang</a> (joint with Frances Sladek)
<li> <a href=https://ellrottlab.org/authors/kellrott/>Kyle Ellrott</a> (undergrad work)
<li> <a href=http://www.cs.ucr.edu/~zliu/>Zheng Liu</a> (now at City of Hope)
<li> <a href=http://www.cs.ucr.edu/~lliu>Lan Liu</a> (now at Google)
<li> <a href=http://www.cs.ucr.edu/~zfu>Zheng Fu</a> (now at Google)
<li> <a href=http://iiis.tsinghua.edu.cn/xiaoj/>Jing Xiao</a> (now at Fullgoal Fund Management Co., Ltd)
<li> <a href=http://www.cs.ucr.edu/~ycao>Eddie Cao</a> (joint with Thomas Girke; now at Novartis Institutes for Biomedical Research or NIBR)
<li> <a href=https://www.researchgate.net/profile/Minzhu_Xie2>Minzhu Xie</a> (now at Hunan Normal University)
<li> <a href=http://www.cs.ucr.edu/~weiw>Bob Wang</a> (now at Google)
<li> <a href=https://www.researchgate.net/profile/Guanqun_Shi>Wilson Shi</a> (now at Google)
<li> <a href=https://www.linkedin.com/in/bin-fang-9559a411>Bin Fang</a> (joint with Frances Sladek; now at GNF)
<li> <a href=http://www.researchgate.net/profile/Jianxing_Feng>Jianxing Feng</a> (now in IT industry, Shanghai)
<li> <a href=https://weililab.org/>Wei Li</a>
<li> <a href=http://www.cs.ucr.edu/~tanaseio/>Olga Tanaseichuk</a> (now at Google)
<li> <a href=http://faculty.bjtu.edu.cn/8743/>Ergude Bao</a> 
(joint with Thomas Girke)
<li> <a href=https://www.linkedin.com/in/shining-ma-402a8523/>Shining Ma</a> (now
Scientist at Amgen; joint with Rui Jiang)
<li> <a href=https://www.linkedin.com/in/ei-wen-yang-078b3153/>Ei-Wen Yang</a> (now Principal Scientist at Kymera Therapeutics)
<li> <a href=https://www.linkedin.com/in/ashraful-arefeen-27511b71>Ashraful Arefeen</a> (now Data Scientist at Walmart Global Tech)
<li> <a href=https://www.linkedin.com/in/dipanshaw>Dipan Shaw</a> (now Deep Learning Engineer at Illumina)
<li> <a href=https://csen.nankai.edu.cn/info/1028/1060.htm>Jianyu Zhou</a>
<li> <a href=https://haochenhc.github.io/>Hao Chen</a>
<li> <a href=https://www.researchgate.net/scientific-contributions/Yuanpeng-Xiong-2149624360>Yuanpeng Xiong</a> (now with a biotech startup)
<li> <a href=https://scholar.google.com/citations?user=gthKG_8AAAAJ&hl=en>Chencheng Xu</a> (now postdoc at KAUST)
<li> <a href=https://scholar.google.com/citations?user=uPBztC0AAAAJ&hl=zh-CN>Zhiwei Hong</a> (now at Intel Labs China)


<hr>
</i>
</body>
</html>
