<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN"
  "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
<head>
<meta name="generator" content="jemdoc, see http://jemdoc.jaboc.net/" />
<meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<link rel="stylesheet" href="jemdoc.css" type="text/css" />
<title>Sriram Sankararaman</title>
<!-- MathJax -->
<script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-MML-AM_CHTML' async>
</script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
	  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
<!-- End MathJax -->
</head>
<body>
<table summary="Table for page layout." id="tlayout">
<tr valign="top">
<td id="layout-menu">
<div class="menu-category">Sriram Sankararaman</div>
<div class="menu-item"><a href="index.html" class="current">Home</a></div>
<div class="menu-item"><a href="cv.pdf">Resume</a></div>
<div class="menu-item"><a href="teaching.html">Teaching</a></div>
<div class="menu-category">Research</div>
<div class="menu-item"><a href="lab.html">Lab</a></div>
<div class="menu-item"><a href="interests.html">Interests</a></div>
<div class="menu-item"><a href="publications.html">Publications</a></div>
<div class="menu-item"><a href="software-data.html">Software&nbsp;and&nbsp;data</a></div>
<div class="menu-item"><a href="press.html">Press</a></div>
<div class="menu-item"><a href="openings.html">Openings</a></div>
<div class="menu-item"><a href="https://sriramlab.dgsom.ucla.edu">Lab&nbsp;website</a></div>
<div class="menu-category">Teaching</div>
<div class="menu-item"><a href="https://web.cs.ucla.edu/~sriram/courses/cm146.winter-2020/html/index.html" class="current">Intro&nbsp;to&nbsp;ML&nbsp;(CM146)</a></div>
<div class="menu-item"><a href="https://web.cs.ucla.edu/~sriram/courses/cm226.fall-2019/html/index.html" class="current">ML4Bio&nbsp;(CM226)</a></div>
<div class="menu-item"><a href="https://web.cs.ucla.edu/~sriram/courses/cm229.spring-2021/html/index.html" class="current">CM229</a></div>
</td>
<td id="layout-content">
<div id="toptitle">
<h1>Sriram Sankararaman</h1>
</div>
<table class="imgtable"><tr><td>
<a href="http://www.cs.ucla.edu/~sriram"><img src="photos/headshot-cropped.jpg" alt="alt text" width="140px" height="160px" /></a>&nbsp;</td>
<td align="left"><p>Professor<br /> <a href="http://www.cs.ucla.edu" target=&ldquo;blank&rdquo;>Computer Science Department</a> <br /><a href="https://medschool.ucla.edu/human-genetics" target=&ldquo;blank&rdquo;>Department of Human Genetics</a><br /><a href="https://www.biomath.ucla.edu" target=&ldquo;blank&rdquo;>Department of Computational Medicine</a><br /><a href="https://bioinformatics.ucla.edu" target=&ldquo;blank&rdquo;>Bioinformatics Interdepartmental Graduate Program</a><br />
University of California, Los Angeles<br />
Email: <i>sriram</i> [@] cs [DOT] ucla [DOT] edu<br />
Lab website: <a href="https://sriramlab.dgsom.ucla.edu/pages/" target=&ldquo;blank&rdquo;>Machine Learning and Genomics Lab</a><br /></p>
</td></tr></table>
<h2>About me</h2>
<p>I am a Professor of Computer Science, Human Genetics, and Computational Medicine at UCLA.

I am broadly interested in problems at the intersection of computer science, statistics, and biomedicine. Here are recent videos that summarize work from our lab: </p>
<ul>
<li><p><a href="https://www.imsi.institute/videos/scaling-statistical-models-to-millions-of-human-genomes" target=&ldquo;blank&rdquo;>Scaling statistical models to millions of genomes@IMSI</a></p>
</li>
<li><p><a href="https://www.youtube.com/watch?v=2P61UtPqpfA" target=&ldquo;blank&rdquo;>Archaic admixture in human history@Radcliffe Institute</a> </p>
</li>
<li><p><a href="https://www.youtube.com/watch?v=8bqU6QsW-sM&amp;t=2s" target=&ldquo;blank&rdquo;>Signals of Ghost Archaic DNA in Present-Day West African Populations</a></p>
</li>
<li><p><a href="https://www.youtube.com/watch?v=vXOyVG-LNoA" target=&ldquo;blank&rdquo;>Recovering signals of ghost archaic introgression in African populations@CARTA</a></p>
</li>
</ul>
<h2>Selected publications</h2>
<ul>
<li><p>A Scalable adaptive quadratic kernel method for interpretable epistasis analysis in complex traits, <i><a href="https://genome.cshlp.org/content/early/2024/08/29/gr.279140.124" target=&ldquo;blank&rdquo;>Genome Research</a></i> (2024)</p>
</li>
<li><p>A scalable and robust variance components method reveals insights into the architecture of gene-environment interactions underlying complex traits,  <i><a href="https://doi.org/10.1016/j.ajhg.2024.05.015" target=&ldquo;blank&rdquo;>The American Journal of Human Genetics</a></i> (2024)

</p>
</li>
<li><p>Deep learning-based phenotype imputation on population-scale biobank data increases genetic discoveries, <i><a href="https://www.biorxiv.org/content/10.1101/2022.08.15.503991v1" target=&ldquo;blank&rdquo;>Nature Genetics</a></i> (2023)</p>
</li>
<li><p>Fast kernel-based association testing of non-linear genetic effects for biobank-scale data, <i><a href="https://www.biorxiv.org/content/10.1101/2022.04.13.488214v1" target=&ldquo;blank&rdquo;>Nature Communications</a></i> (2023)</p>
</li>
<li><p>The lingering effects of Neanderthal introgression on human complex traits, <i><a href="https://elifesciences.org/articles/80757" target=&ldquo;blank&rdquo;>eLife</a></i> (2023)

</p>
</li>
<li><p>Leveraging family data to design Mendelian Randomization that is provably robust to population stratification, <i><a href="https://genome.cshlp.org/content/early/2023/05/17/gr.277664.123" target=&ldquo;blank&rdquo;>Genome Research</a></i> (2023) <br />

</p>
</li>
<li><p>Inferring population structure in biobank-scale genomic data, <i><a href="https://www.sciencedirect.com/science/article/pii/S0002929722000660" target=&ldquo;blank&rdquo;>The American Journal of Human Genetics</a></i> (2022) </p>
</li>
<li><p>Marginal Contribution Feature Importance &ndash; an axiomatic approach for the natural case, <i><a href="https://proceedings.mlr.press/v139/catav21a.html" target=&ldquo;blank&rdquo;>ICML</a></i> (2021) </p>
</li>
<li><p>Advancing admixture graph estimation via maximum likelihood network orientation, <i><a href="https://academic.oup.com/bioinformatics/article/37/Supplement_1/i142/6319689" target=&ldquo;blank&rdquo;>Bioinformatics (special issue of ISMB/ECCB)</a></i> (2021) 



</p>
</li>
<li><p>Quantifying the contribution of dominance effects to complex trait variation in biobank-scale data, <i><a href="https://linkinghub.elsevier.com/retrieve/pii/S0002-9297(21)00102-6" target=&ldquo;blank&rdquo;>The American Journal of Human Genetics</a></i> (2021)



</p>
</li>
<li><p>Efficient variance components analysis across millions of genomes, <i><a href="https://www.nature.com/articles/s41467-020-17576-9" target=&ldquo;blank&rdquo;>Nature Communications</a></i> (2020) 


</p>
</li>
<li><p>Scalable probabilistic PCA for large-scale genetic variation data, <i><a href="https://doi.org/10.1371/journal.pgen.1008773" target=&ldquo;blank&rdquo;>PLoS Genetics</a></i> (2020) </p>
</li>
<li><p>Recovering signals of ghost archaic introgression in African populations, <i><a href="https://advances.sciencemag.org/content/6/7/eaax5097" target=&ldquo;blank&rdquo;>Science Advances</a></i> (2020) </p>
</li>
<li><p>The combined landscape of Denisovan and Neanderthal ancestry in present-day humans, <i><a href="https://doi.org/10.1016/j.cub.2016.03.037" target=&ldquo;blank&rdquo;>Current Biology</a></i> (2016) </p>
</li>
<li><p>Genetic and socioeconomic study of mate choice in Latinos reveals novel assortment patterns, <i><a href="https://doi.org/10.1073/pnas.1501741112" target=&ldquo;blank&rdquo;>PNAS</a></i> (2016) </p>
</li>
<li><p>The genomic landscape of Neanderthal ancestry in present-day humans, <i><a href="https://www.nature.com/articles/nature12961" target=&ldquo;blank&rdquo;>Nature</a></i> (2014) 

</p>
</li>
<li><p>Genomic privacy and limits of individual detection in a pool, <i><a href="https://www.nature.com/articles/ng.436" target=&ldquo;blank&rdquo;>Nature genetics</a></i> (2009) </p>
</li>
<li><p>Estimating local ancestry in admixed populations, <i><a href="https://www.cell.com/ajhg/fulltext/S0002-9297(08)00079-7" target=&ldquo;blank&rdquo;>The American Journal of Human Genetics</a></i> (2008) </p>
</li>
</ul>
<h2>Honors and Awards</h2>
<ul>
<li><p>NSF Career Award (2020)</p>
</li>
<li><p>Microsoft Investigator Fellow (2019)</p>
</li>
<li><p>Northrop Grumman Excellence in Teaching Award, UCLA (2019)</p>
</li>
<li><p>Okawa Foundation Research Grant (2017)</p>
</li>
<li><p>UCLA Hellman Fellow (2017)</p>
</li>
<li><p>Alfred P. Sloan Fellow (2017)</p>
</li>
<li><p>NIH Pathway to Independence Award (2014)</p>
</li>
<li><p>Fellow, Simons Institute for the Theory of Computing, Berkeley (2014)</p>
</li>
<li><p>Harvard Science of the Human Past fellow (2012)</p>
</li>
<li><p>Visvesvaraya medal for highest CGPA in the graduating class, IIT Madras (2004)</p>
</li>
</ul>
</td>
</tr>
</table>
</body>
</html>
